cleverhans/experimental/certification/nn.py

```diff
@@ -54,15 +52,16 @@ def __init__(self, net_weights, net_biases, net_layer_types, input_shape=None):
       current_num_rows = input_shape[0]
       current_num_columns = input_shape[1]
       current_num_channels = input_shape[2]
+    if(cnn_params is not None):
```

Comment:
The problem with this approach is that stride and padding will be the same for all convolutions in the network, which in practice not the case for real neural networks.
Padding and stride should be defined per convolution layer

Another suggestions is following.
Having tuple of values with different meaning could be not very clear for readers of the code.
Alternatives are either to use [namedtuple](https://docs.python.org/2/library/collections.html#collections.namedtuple) or use two different arguments, one for stride and one for padding.